aioftp/client.py

```diff
@@ -338,6 +341,188 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        def parse_rw(s):
+            mode = 0
+
+            if s[0] == 'r':
+
+                mode |= 4
+
+            elif s[0] != '-':
+
+                raise ValueError
+
+            if s[1] == 'w':
+
+                mode |= 2
+
+            elif s[1] != '-':
+
+                raise ValueError
+
+            return mode
+
+        mode = 0
+        mode |= parse_rw(s[0:2]) << 6
+        mode |= parse_rw(s[3:5]) << 3
+        mode |= parse_rw(s[6:8])
+
+        if s[2] == 's':
+
+            mode |= 0o4100
+
+        elif s[2] == 'x':
+
+            mode |= 0o0100
+
+        elif s[2] != '-':
+
+            raise ValueError
+
+        if s[5] == 's':
+
+            mode |= 0o2010
+
+        elif s[5] == 'x':
+
+            mode |= 0o0010
+
+        elif s[5] != '-':
+
+            raise ValueError
+
+        if s[8] == 't':
+
+            mode |= 0o1000
+
+        elif s[8] == 'x':
+
+            mode |= 0o0001
+
+        elif s[8] != '-':
+
+            raise ValueError
+
+        return mode
+
+    @staticmethod
+    def parse_ls_date(s):
+        """
+        Parsing dates from the ls unix utility
+        ("Nov 18  1958" and "Nov 18 12:29")
+
+        :param s: ls date
+        :type s: :py:class:`str`
+
+        :rtype: :py:class:`str`
+        """
+        if not (s[3] == ' ' and s[6] == ' '):
+
+            raise ValueError
+
+        month = "{0:02d}".format(time.strptime(s[:3], "%b").tm_mon)
+        day = s[4:6]
+
+        if day[0] == ' ':
+
+            day = '0' + day[1:]
+
+        if s[9] == ':':
+
+            year = "0000"
+            hour_min = s[7:9] + s[10:12]
+
+        else:
+
+            hour_min = "0000"
+
+            if s[6] != ' ':
+
+                raise ValueError
+
+            year = s[8:12]
+
+        result = year + month + day + hour_min
+
+        if not result.isdigit():
+
+            raise ValueError
+
+        return result + "00"
+
+    def parse_list_line(self, b):
+        """
+        Attempt to parse a LIST line(similar to unix ls utility)
+
+        :param b: LIST line
+        :type b: :py:class:`bytes` or :py:class:`str`
+
+        :return: (path, info)
+        :rtype: (:py:class:`pathlib.PurePosixPath`, :py:class:`dict`)
+        """
+        if isinstance(b, bytes):
+
+            s = str.rstrip(bytes.decode(b, encoding="utf-8"))
+
+        else:
+
+            s = b
+
+        info = {}
+
+        if s[0] == '-':
+
+            info["type"] = "file"
+
+        elif s[0] == 'd':
+
+            info["type"] = "dir"
+
+        else:
+
+            info["type"] = "unknown"
```

Comment:
Don't really know what user will do with «unknown»? And what type of object is this?